src/api_manager/context/request_context.cc

```diff
@@ -264,6 +267,38 @@ void RequestContext::FillLogMessage(service_control::ReportRequestInfo *info) {
   }
 }
 
+void RequestContext::FillHttpHeaders(service_control::ReportRequestInfo *info,
+                                     Response *response) {
+  auto serverConfig = service_context_->config()->server_config();
+  if (serverConfig->has_service_control_config()) {
+    auto request_headers =
+        serverConfig->service_control_config().log_request_headers();
+    std::vector<std::string> req_headers;
+    split(request_headers, kHttpHeadersDelimeter, &req_headers);
+    for (auto &header : req_headers) {
+      auto key = trim(header);
+      std::string header_value;
+      if (request_->FindHeader(key, &header_value)) {
+        info->request_headers =
+            info->request_headers + key + "=" + header_value + ";";
+      }
+    }
+
+    auto response_headers =
+        serverConfig->service_control_config().log_response_headers();
+    std::vector<std::string> resp_headers;
+    split(response_headers, kHttpHeadersDelimeter, &resp_headers);
+    for (auto &header : resp_headers) {
+      auto key = trim(header);
+      std::string header_value;
+      if (response->FindHeader(key, &header_value)) {
```

Comment:
This is O(n * m) operation, it is too slow.   Maybe we can convert the resp_headers into a unordered_set first,  and pass in the set to FindHeader to sequentail search one by one.  so it will be O(n),  n is number of headers.